src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/PrisonPersonService.kt

```diff
@@ -62,8 +62,12 @@ class PrisonPersonService(
       }
   }
 
-  fun getProfileDetails(offenderNo: String): PrisonerProfileDetailsResponse =
-    bookingRepository.findAllByOffenderNomsId(offenderNo)
+  fun getProfileDetails(offenderNo: String): PrisonerProfileDetailsResponse {
+    if (!offenderRepository.existsByNomsId(offenderNo)) {
```

Comment:
odd that the repository doesn't provide a `getReferenceById` or similar method that already throws the entity not found